modules/mod_ginger_base/controllers/controller_auth.erl

```diff
@@ -48,6 +50,39 @@ process_post(Req, State = #state{mode = new}) ->
     {ok, Id} = mod_signup:signup([{email, Email}], [{identity, Identity}], RequestConfirm, Context),
     ok = mod_signup:request_verification(Id, Context),
     {{halt, 204}, Req1, State};
+process_post(Req, State = #state{mode = reset_password}) ->
+    Context = State#state.context,
+    {Body, Req1} = wrq:req_body(Req),
+    #{u := Username,
+      secret := Secret,
+      password1 := Password1,
+      password2 := Password2} = jsx:decode(Body, [return_maps, {labels, atom}]),
+    PasswordMinLength = z_convert:to_integer(
+                          m_config:get_value(mod_ginger_base, password_min_length, "6", Context)),
+    case {Password1,Password2} of
+        {A,_} when length(A) < PasswordMinLength ->
```

Comment:
We should check against `m.config.mod_admin_identity.password_regex`, too. (Zotonic checks against this in templates.)